Excercise_4/assets/scripts/script.js

```diff
@@ -84,121 +107,125 @@ const showCarousel = (dataList, indexCarousel) => {
-		const slide = slides[i].children;
-		for (let index = 0; index < numOfSlide[i]; index++) {
-			const ratio = currentTranslate[i][index] / oldOffset;
-			currentTranslate[i][index] = newOffset * ratio;
-			slide[index].style.transform = `translateX(${currentTranslate[i][index]}px)`;
+	};
+
+	const moveToSlide = (carouselIndex, moveOffset) => {
+		const slideChildren = slides[carouselIndex].children;
+		for (let index = 0; index < slideChildren.length; index++) {
+			slideChildren[index].style.transform = `translateX(${
+				currentTranslate[carouselIndex][index] + moveOffset
+			}px)`;
+			currentTranslate[carouselIndex][index] = currentTranslate[carouselIndex][index] + moveOffset;
 		}
-	}
+		const index = currentSlide[carouselIndex] % numOfSlide[carouselIndex];
+		slides[carouselIndex].children[index].style.transform = `translateX(${
+			currentTranslate[carouselIndex][index] - moveOffset * numOfSlide[carouselIndex]
+		}px)`;
+		currentTranslate[carouselIndex][index] =
+			currentTranslate[carouselIndex][index] - moveOffset * numOfSlide[carouselIndex];
+	};
+
+	return {
+		makeSlide: function () {
+			moveOffset = slides[0].children[0].offsetWidth + MARGIN_RIGHT;
+
+			for (let index = 0; index < slides.length; index++) {
+				numOfSlide.push(slides[index].children.length);
+				slides[index].style.width = moveOffset * numOfSlide[index] + 'px';
+
+				currentSlide.push(0);
+				currentTranslate[index] = [];
+
+				for (let i = 0; i < numOfSlide[index]; i++) {
+					currentTranslate[index].push(0);
+				}
+			}
+		},
+		handleWindowResize: function () {
+			window.addEventListener('resize', () => {
+				const oldMoveOffset = moveOffset;
+				let newMoveOffset = MARGIN_RIGHT;
+				if (carouselContainer.offsetWidth < MOBILE_CONTAINER_WIDTH) {
+					newMoveOffset += MOBILE_ITEM_WIDTH;
+				} else {
+					newMoveOffset += OTHER_DEVICE_ITEM_WIDTH;
+				}
+				changeWidthForCarouselItem(oldMoveOffset, newMoveOffset);
+				moveOffset = newMoveOffset;
+				changeMenuList(); ///
+			});
+		},
+		handleClickNextCarousel: function (carouselIndex) {
+			currentSlide[carouselIndex]--;
+			if (currentSlide[carouselIndex] < 0) {
+				currentSlide[carouselIndex] = numOfSlide[carouselIndex] - 1;
+			}
+			moveToSlide(carouselIndex, moveOffset);
+		},
+		handleClickPrevCarousel: function (carouselIndex) {
+			moveToSlide(carouselIndex, -moveOffset);
+			currentSlide[carouselIndex]++;
+		},
+		handleTouchStart: function (carouselIndex) {
+			slides[carouselIndex].addEventListener('touchstart', (event) => {
+				startX = event.changedTouches[0].clientX;
+			});
+		},
+		handleTouchEnd: function (carouselIndex) {
+			slides[carouselIndex].addEventListener('touchend', (event) => {
+				endX = event.changedTouches[0].clientX;
+
+				if (endX - startX > 0) handleClickNextCarousel(carouselIndex);
+				else if (endX - startX < 0) handleClickPrevCarousel(carouselIndex);
+			});
+		},
+	};
+}
+
+const carouselMaker = makeCarousel();
```

Comment:
what do you think about handleCarouselEvent ?

Reply:
nopee~~, it is not only handle carousel event but also init carousel or do something relevant